test/Interop/Cxx/value-witness-table/witness-layout-opts-irgen.swift

```diff
@@ -0,0 +1,20 @@
+// RUN: %target-swift-emit-ir %s -I %S/Inputs -enable-cxx-interop | %FileCheck %s
+
+import WitnessLayoutOpts
+
+protocol Dummy { }
+
+// CHECK-DAG: @"$sBi8_WV" = external global i8*, align 8
+// CHECK-DAG: @"$sSo5EmptyVMn" = linkonce_odr hidden constant <{ i32, i32, i32, i32, i32, i32, i32, i32 }> <{ i32 131089, i32 trunc (i64 sub (i64 ptrtoint (<{ i32, i32, i32 }>* @"$sSoMXM" to i64), i64 ptrtoint (i32* getelementptr inbounds (<{ i32, i32, i32, i32, i32, i32, i32, i32 }>, <{ i32, i32, i32, i32, i32, i32, i32, i32 }>* @"$sSo5EmptyVMn", i32 0, i32 1) to i64)) to i32), i32 trunc (i64 sub (i64 ptrtoint ([6 x i8]* @5 to i64), i64 ptrtoint (i32* getelementptr inbounds (<{ i32, i32, i32, i32, i32, i32, i32, i32 }>, <{ i32, i32, i32, i32, i32, i32, i32, i32 }>* @"$sSo5EmptyVMn", i32 0, i32 2) to i64)) to i32), i32 trunc (i64 sub (i64 ptrtoint (%swift.metadata_response (i64)* @"$sSo5EmptyVMa" to i64), i64 ptrtoint (i32* getelementptr inbounds (<{ i32, i32, i32, i32, i32, i32, i32, i32 }>, <{ i32, i32, i32, i32, i32, i32, i32, i32 }>* @"$sSo5EmptyVMn", i32 0, i32 3) to i64)) to i32), i32 trunc (i64 sub (i64 ptrtoint ({ i32, i32, i16, i16, i32 }* @"$sSo5EmptyVMF" to i64), i64 ptrtoint (i32* getelementptr inbounds (<{ i32, i32, i32, i32, i32, i32, i32, i32 }>, <{ i32, i32, i32, i32, i32, i32, i32, i32 }>* @"$sSo5EmptyVMn", i32 0, i32 4) to i64)) to i32), i32 0, i32 2, i32 0 }>, section "__TEXT,__const", align 4
+// CHECK-DAG: @"$sSo5EmptyVMf" = linkonce_odr hidden constant <{ i8**, i64, <{ i32, i32, i32, i32, i32, i32, i32, i32 }>* }> <{ i8** @"$sBi8_WV", i64 512, <{ i32, i32, i32, i32, i32, i32, i32, i32 }>* @"$sSo5EmptyVMn" }>, align 8
```

Comment:
I'm not entirely sure that this is the correct (or portable) way to test this. It would probably be sufficient just to test line 9, 14, and 19.